src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultActionModelBuilder.cs

```diff
@@ -256,6 +257,15 @@ private bool IsIDisposableMethod(MethodInfo methodInfo, TypeInfo typeInfo)
             AddRange(actionModel.ActionConstraints, attributes.OfType<IActionConstraintMetadata>());
             AddRange(actionModel.Filters, attributes.OfType<IFilter>());
 
+            foreach (var authorizeAttribute in attributes.OfType<AuthorizeAttribute>())
+            {
+                actionModel.Filters.Add(new AuthorizeFilter
+                {
+                    Policy = authorizeAttribute.Policy,
+                    Roles = authorizeAttribute.Roles?.Split(','),
+                });
+            }
```

Comment:
Need to see a unit test for this and also for controllermodelbuilder when that's fixed too.